### ISSUE TITLE
Nightlies: Please remove old versioned nightly builds

### ISSUE THREAD

**Issue by seberg** [matplotlib/matplotlib#27243]
TL;DR: Can you please remove all releases from https://anaconda.org/scientific-python-nightly-wheels/matplotlib/files which are not `.dev0`?
I currently seem to get the old releases due to their higher version number, defeating the purpose :).

(I thought there was an auto-cleanup of very old releases, but it doesn't really matter in practice.)

I guess matpltolib ran into a similar thing as we did in NumPy, switching to meson a month ago or so?  In either case, it seems nightlies are now uploaded with `.dev0` while previously they were uploaded with `.dev<#commits>...` or similar.

Now, I can't say I don't like the fact that the nightlies stay around a bit longer with a clearer version than just `.dev0`, but it doesn't matter much.

PS: Are the MacOSX 11_0 wheels missing or is that intentional?

EDIT: Ping @ksunden, I just realized you are probably the one who has access.

**Comment by ksunden** [comment]
Should be good, I think that we are dependent on mesonpy supporting dynamic versions to get the more informative names (see https://github.com/mesonbuild/meson-python/issues/159)

I think the macos wheels is related to the meson transition, do you know how numpy gets those wheels? I don't see anything obvious, best guess is our `MACOSX_DEPLOYMENT_TARGET` env var is interfering in a way that was overridden when we used setuptools and not when we are using meson...

Anywho, I'll close the main issue here, and will open a new one with discussion on the macosx things more directly, as that is orthogonal (but I will acknowledge that the wheels would not install on my mac, so it does need to be sorted out before we release a meson based release...)

**Comment by seberg** [comment]
Thanks!  Not sure if I can be of direct help.  I think we have a very similar setup to yours, but are additionally using CirrusCI native machiens for those wheel builds: https://github.com/numpy/numpy/blob/main/tools/ci/cirrus_wheels.yml
The free tier might be enough for releases and occasional nightlies, NumPy is just spending a bit of money on it these days, because it is tedious to get arm machines otherwise.

**Comment by seberg** [comment]
Ping @andyfaff who did the heavy lifting for NumPy's CirrusCI, case he has some time/tip for you.

**Comment by andyfaff** [comment]
What kind of advice is required, it wasn't quite clear? I know how to setup for cirrus-ci reasonably well and can help there. I don't know how older versions are removed from the nightly wheels bucket, Ralf did that manually for some of scipy's. I think there may even be an automated scraper that removes some older versions.
If it's about how wheels get named, e.g. why is 10_12 in the name, then I figure things out by trial and error. If this is the question then the following environment variables may be useful in a cibuildwheel run for arm64:
```shell
    CIBW_ENVIRONMENT: >
      MACOSX_DEPLOYMENT_TARGET=12.0
      _PYTHON_HOST_PLATFORM="macosx-12.0-arm64" 
```
It produces wheel names like `scipy-1.12.0.dev0-cp311-cp311-macosx_12_0_arm64.whl`.

If the question is about commit hashes in the wheel name being lost, then I think that occurred as part of the setup->meson transition.

**Comment by andyfaff** [comment]
(I think we used both of those env vars because we only wanted to distribute arm64 wheels for macosx >=12.0, something was broken with 11).

**Comment by ksunden** [comment]
@andyfaff @seberg I think we tracked down what was happening, `setuptools` (via `wheel`) overrode the env var for ARM (because xcode overrides the build time setting on ARM, it seems)

See https://github.com/mesonbuild/meson-python/issues/524 for more info, but the main point is that mesonpy doesn't do that (but cibuildwheel at least does set 11.0 as the default on arm builds, though it doesn't override the env var if set, which is why numpy gets 11.0 for arm and 10.9 for x86_64)
